pkg/network/ovn_kubernetes.go

```diff
@@ -377,13 +391,29 @@ func renderOVNKubernetes(conf *operv1.NetworkSpec, bootstrapResult *bootstrap.Bo
 	var manifestSubDir string
 	manifestDirs := make([]string, 0, 2)
 	manifestDirs = append(manifestDirs, filepath.Join(manifestDir, "network/ovn-kubernetes/common"))
+
+	productFlavor := "self-hosted"
 	if bootstrapResult.OVN.OVNKubernetesConfig.HyperShiftConfig.Enabled {
-		manifestSubDir = "network/ovn-kubernetes/managed"
-		manifestDirs = append(manifestDirs, filepath.Join(manifestDir, manifestSubDir))
-	} else {
-		manifestSubDir = "network/ovn-kubernetes/self-hosted"
-		manifestDirs = append(manifestDirs, filepath.Join(manifestDir, manifestSubDir))
+		productFlavor = "managed"
+	}
+	manifestSubDirBasePath := filepath.Join("network/ovn-kubernetes", productFlavor)
+
+	manifestDirs = append(manifestDirs, filepath.Join(manifestDir, manifestSubDirBasePath, "common"))
+
+	// choose the YAMLs based on the target zone mode (4.14 only) TODO: starting from 4.15, support multizone only
+	manifestSubDir = filepath.Join(manifestSubDirBasePath, "multi-zone-interconnect") // default is multizone
+	if targetZoneMode.zoneMode == zoneModeSingleZone {
+		// non-default, internal use only; this is selected in the first phase of an upgrade from a
+		// non-interconnect version (< 4.14) to an interconnect version (>= 4.14)
+		manifestSubDir = filepath.Join(manifestSubDirBasePath, "single-zone-interconnect")
+	} else if isMigrationToMultiZoneAboutToStartOrOngoing(bootstrapResult.OVN, &targetZoneMode) {
+		// intermediate step when converting from single zone to multizone; this is selected
+		// in the second phase of an upgrade from a non-interconnect version (< 4.14)
+		// to an interconnect version (>= 4.14)
+		manifestSubDir = filepath.Join(manifestSubDirBasePath, "multi-zone-interconnect-tmp")
```

Comment:
oh I see because some symlink to the single zone folder for legacy master stuff...nice

Reply:
Yeah, we keep the ovnk master from single-zone and, in case of hypershift, also the sbdb route. This allows us to have a fully working cluster in hypershift all throughout the upgrade, since the route is up as long as the multizone cluster is not 100% deployed, and we can get rid of the hypershift statefulset for master. In standalone openshift, we follow the same upgrade path and we get rid of the master daemonset, which is replaced by the control-plane deployment.